internal/plugin/server.go

```diff
@@ -113,27 +94,53 @@ func NewNvidiaDevicePlugin(config *spec.Config, kubeletSocket string, resourceMa
 	return &plugin, nil
 }
 
-func (plugin *NvidiaDevicePlugin) initialize() {
+// getMPSDaemonAndRoot returns the MPS daemon and root for the specified resource manager.
+// TODO: We should return a type here to manage MPS specifics.
+func (o *options) getMPSDaemon(resourceManager rm.ResourceManager) (*mps.Daemon, mps.Root, error) {
+	if o.config.Sharing.SharingStrategy() != spec.SharingStrategyMPS {
+		return nil, "", nil
+	}
+
+	// TODO: It might make sense to pull this logic into a resource manager.
+	for _, device := range resourceManager.Devices() {
+		if device.IsMigDevice() {
+			return nil, "", errors.New("sharing using MPS is not supported for MIG devices")
+		}
+	}
+	mpsDaemon := mps.NewDaemon(resourceManager, mps.ContainerRoot)
+	mpsHostRoot := mps.Root(*o.config.Flags.CommandLineFlags.MpsRoot)
+
+	return mpsDaemon, mpsHostRoot, nil
+}
+
+// getPluginSocketPath returns the socket to use for the specified resource.
+func getPluginSocketPath(resource spec.ResourceName) string {
+	_, name := resource.Split()
+	pluginName := "nvidia-" + name
+	return filepath.Join(pluginapi.DevicePluginPath, pluginName) + ".sock"
+}
+
+func (plugin *nvidiaDevicePlugin) initialize() {
 	plugin.server = grpc.NewServer([]grpc.ServerOption{}...)
 	plugin.health = make(chan *rm.Device)
 	plugin.stop = make(chan interface{})
 }
 
-func (plugin *NvidiaDevicePlugin) cleanup() {
+func (plugin *nvidiaDevicePlugin) cleanup() {
 	close(plugin.stop)
 	plugin.server = nil
 	plugin.health = nil
 	plugin.stop = nil
 }
 
 // Devices returns the full set of devices associated with the plugin.
-func (plugin *NvidiaDevicePlugin) Devices() rm.Devices {
+func (plugin *nvidiaDevicePlugin) Devices() rm.Devices {
 	return plugin.rm.Devices()
 }
 
 // Start starts the gRPC server, registers the device plugin with the Kubelet,
 // and starts the device healthchecks.
-func (plugin *NvidiaDevicePlugin) Start() error {
+func (plugin *nvidiaDevicePlugin) Start(kubeletSocket string) error {
```

Comment:
I see -- its' not stored, but rather passed in to `Start()` now...

Reply:
Yes, this is how I had it for the original change to allow an empty string for testing basic functionality without a full cluster. After a round of reviews, I think @tariq1890 suggested we make it a property instead. After looking at this file again from the perspective of this cleanup, I am back to thinking that it makes more sense to pass it as an argument.

Reply:
Your choice. I don't feel strongly.

Reply:
I prefer the way it is now. If we see that we need to connect to mulitple kubelet sockets, we can make that change later.